simpsom/neighborhoods.py

```diff
@@ -8,15 +8,22 @@
 class Neighborhoods:
     """ Container class with functions to calculate neihgborhoods. """
 
-    def __init__(self, xp: ModuleType = None) -> None:
+    def __init__(self, xp: ModuleType, xx: np.ndarray, yy: np.ndarray, pbc_func: Union[Callable, None]) -> None:
```

Comment:
I'm not against having xx, yy, and the other arguments set as attributes of a neighborhood object, but in `network.py` then you instantiate a new Neighborhoods object for each training and then call `neighborhood_caller` once, defeating the purpose of having it as a class.